modules/fluid_properties/include/userobjects/SinglePhaseFluidProperties.h

```diff
@@ -328,6 +328,71 @@ class SinglePhaseFluidProperties : public FluidProperties
                               Real & de_dp,
                               Real & de_dT) const;
 
+  /**
+   * Determines (p,T) from (v,e) using Newton Solve in 2D
+   * Useful for inversion of state variables
+   *
+   * @param[in] v specific volume (m^3 / kg)
+   * @param[in] e specific internal energy (J / kg)
+   * @param[in] p0 initial guess for pressure (Pa / kg)
+   * @param[in] T0 initial guess for temperature (K)
+   * @param[out] fluid pressure (Pa / kg)
+   * @param[out] Temperature (K)
+   */
+  virtual void p_T_from_v_e(const Real & v,
```

Comment:
there's nothing specific about wanting to use maths to perform an inversion. 
in the vast majority of fluid properties, these conversions are not analytic, which is also why support for (v,e) is so limited outside of the FP repos.

Reply:
if we dont have this here, we're going to have the same thing in all the downstream classes that are not over-simplified

Reply:
My main reservation is that users of all fluid properties would now have the parameters "tolerance", "T_initial_guess", and "p_initial_guess", which are usually irrelevant for them, which is confusing to the user (do I need to set this? what does this do?). Also developers may use one of these solve methods, when they have more direct and efficient options to them. However, I'm fine with keeping this. I originally had thought this was used only in `TabulatedFluidProperties`, but now I see the new default implementations (which I'm guessing are needed indirectly in `TabulatedFluidProperties`).

Reply:
those parameters are in the `Variable set conversions Newton solve` group.
I ll add docs in SFP.md

afaik there are no closed forms to convert between variable sets in the non-easy direction. The developers will be happy to find these routines rather than implement their own.

Reply:
ugh we dont have that page. I ll upload all other revisions and work on writing it